R/bind-rows.R

```diff
@@ -65,7 +65,9 @@ bind_rows <- function(..., .id = NULL) {
     check_string(.id)
 
     if (!is_named(dots)) {
-      names(dots) <- seq_along(dots)
+      vec_present_names <- rlang::have_name(dots)
+      vec_missing_indices <- which(vec_present_names == FALSE)
+      names(dots)[vec_missing_indices] <- vec_missing_indices
```

Comment:
```suggestion
      # Replace `NA` or `""` names with their index,
      # but leave existing names in place (#7100)
      dots_with_names <- rlang::have_name(dots)
      dots_without_names <- which(!dots_with_names)
      names(dots)[dots_without_names] <- as.character(dots_without_names)
```


- Added a comment for future us, referencing the original issue if we need to refer back to it
- I like `dots_with_names` and `dots_without_names` as clear names here
- Just for clarity of _intent_, I like seeing the explicit `as.character(dots_without_names)` call

Reply:
This all makes sense and I agree. Much better clarity.